pynopegl-utils/pynopegl_utils/viewer/__init__.py

```diff
@@ -212,6 +212,20 @@ def _select_scene(self, index):
 
         self._load_current_scene()
 
+    def _test_settings(self):
+        profile_id = self._config.get("export_profile")
+        profile = ENCODE_PROFILES[profile_id]
+
+        # Check if extension is consistent with format
+        filename = self._export_filename_text.property("text")
+        filename = _uri_to_path(filename)
+        filepath = Path(filename)
+        extension = f".{profile.format}"
+        if filepath.suffix.lower() != extension:
+            filepath = filepath.parent / f"{filepath.stem}{extension}"
+            self._export_filename_text.setProperty("text", filepath.as_posix())
+            self._set_export_filename()
```

Comment:
Shouldn't we raise an error instead ? Changing the extension automatically, without notifying the user, could lead to the exporter overwriting an existing file.

Reply:
Added a message dialog when it happens

Reply:
Thanks. The text box is cropped though, see: https://0x5c.me/screenshots/1695108960_grim.png